capstone/scripts/ingest_by_manifest.py

```diff
@@ -184,39 +184,54 @@ def process_volume(vol_entry_bytestring):
             if volume:
                 if is_same_complete_volume(volume, vol_entry, bucket, queues, vol_entry['barcode']):
                     return False
-                else:
-                    volume.delete()
-            volume = VolumeXML(orig_xml=ingest_storage.contents(volmets_path), barcode=vol_entry['barcode'], s3_key=volmets_path)
-            volume.save()
+            else:
+                volume = VolumeXML(barcode=vol_entry['barcode'])
 
+            volume.orig_xml = ingest_storage.contents(volmets_path)
+            volume.s3_key = volmets_path
+            volume.save()
+            
+            existing_case_ids = set(CaseXML.objects.filter(volume=volume).values_list('case_id', flat=True))
```

Comment:
Nit: this is more idiomatic in Django if you use `volume.case_xmls`, which means the same thing as `CaseXML.objects.filter(volume=volume)` but is easier to read. In general if you have one object loaded in and want to fetch related objects, it's more natural in Django to use the first object and work from there.

(See https://docs.djangoproject.com/en/dev/topics/db/queries/#related-objects if you want to dig in more on object relationships in Django ...)